src/Algebra/Construct/Zero.agda

```diff
@@ -17,47 +19,51 @@ open import Level using (Level)
 
 module Algebra.Construct.Zero {c ℓ : Level} where
 
+open import Algebra.Bundles.Raw
+  using (RawMagma)
 open import Algebra.Bundles
-open import Data.Unit.Polymorphic
+  using (Magma; Semigroup; Band)
 
 ------------------------------------------------------------------------
--- Raw bundles
+-- re-export those algebras which are both initial and terminal
 
-rawMagma : RawMagma c ℓ
-rawMagma = record { Carrier = ⊤ ; _≈_ = λ _ _ → ⊤ }
+open import Algebra.Construct.Terminal public
+  hiding (rawMagma; magma; semigroup; band)
```

Comment:
I'm a little bit confused about what the aim of this deprecation is. 

You're hiding the definitions `rawMagma` etc, but then using them qualified and then marking them deprecated. To me this looks like the worst of all worlds :sweat_smile: backwards compatibility is broken and they're still being flagged as deprecated!

What is the desired final state of this module is in the future after any deprecations have eventually been removed? Should it have any of its own declarations or should it just re-export the contents of `Terminal`? If so do we want to deprecate the module itself?

Reply:
Thanks for seeking clarification. let me try to express what I had been trying to achieve... and how precisely to (be able to) do so in Agda may be where I am coming unstuck. 

`Zero` defines structures/bundles which are `Terminal`, for those algebras where the two are identified. 
But it *also* defines, implicitly, by dint of the module name, the `Terminal` `rawMagma`, `magma` and `semigroup` *as if* they were `Zero` objects: they're not. 

But the `Terminal` ones which *are* zero objects should be re-exported by this module (there will, even if for backwards compatibility only, be reasons to view those terminal objects as zero objects, and only want to import them from this module, not from `Terminal`), precisely because there are terminal pbjects which are zero ones.

The 'others' which this module used to define, are included for backwards compatibility, but deprecated, because `Zero.magma` etc. aren't correct mathematically. So I provided definitions for them, and then deprecated them. 

In the future, I don't see this module as being deprecated altogether, even though it will consist purely of re-exporting a *subset* of the definitions in `Terminal`. That's what I had hoped to achieve by the `hiding` directive, but perhaps I don't understand the semantics of it properly to achieve that effect

It makes perfect sense (to me, at least) for `Zero` to re-export those `Terminal` algebras which are in fact also `Zero` objects. 

`Initial` is then the place to define the actual initial objects for `Magma` and `Semigroup`. 

I hope this makes sense!

Reply:
Don't know if this resolves your concerns. There may be something I'm still not understanding...

Reply:
Thank you for the detailed explanation. I now understand :smile: